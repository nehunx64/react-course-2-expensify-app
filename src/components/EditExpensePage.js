import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        //console.log('updated', expense);
        this.props.editExpense(this.props.match.params.id, expense);
        this.props.history.push('/');
    };
    onClick = () => {
        this.props.removeExpense({ id: this.props.match.params.id});
        this.props.history.push('/');               
    }

    render() {
        return (
            <div>
            <ExpenseForm
                expense={ this.props.expense }
                onSubmit={ this.onSubmit } />
            <br />
            <button onClick={ this.onClick }>Remove</button>
        </div>
        )
    };
}
// const EditExpensePage = (props) => {
//     return (
//         <div>
//             <ExpenseForm
//                 expense={props.expense}
//                 onSubmit={(expense) => {
//                     console.log('updated', expense);
//                     props.dispatch(editExpense(props.match.params.id, expense));
//                     props.history.push('/');
//             }} />
//             <br />
//             <button onClick={ () => {
//                 props.dispatch(removeExpense({ id:props.match.params.id}));
//                 props.history.push('/');               
//             }}>Remove</button>
//         </div>
//     );
// };

const mapDispatchToProps = (dispatch, props) => ({
    removeExpense: (id) => dispatch(removeExpense({id})),
    editExpense: (id, expense) => dispatch(editExpense(id, expense))
});

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((ex) => ex.id === props.match.params.id)
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

