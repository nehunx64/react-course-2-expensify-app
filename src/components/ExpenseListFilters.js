import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }

    render() {
        return (
            <div>
                <input type="text" 
                    value={this.props.filters.text}
                    onChange = {(e) => {
                        this.props.dispatch(setTextFilter(e.target.value));
                        // console.log(e.target.value);
                    }} />
                <select 
                    value={this.props.filters.sortBy}
                    onChange = {(e) => {
                    if (e.target.value === 'date') {
                        this.props.dispatch(sortByDate());
                    } else if (e.target.value === 'amount') {
                        this.props.dispatch(sortByAmount());
                    }
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseListFilters);