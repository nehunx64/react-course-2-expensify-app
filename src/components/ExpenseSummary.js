import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div>
    {
        props.count === 1 ? (
            <h1>Viewing {props.count} expense totalling {numeral(props.sum / 100).format('$0,0.00')}</h1>
        ) : (
            props.count > 1 ? (
                <h1>Viewing {props.count} expenses totalling {numeral(props.sum / 100).format('$0,0.00')}</h1>
            ) : ''
        )
    }
    </div>
);

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        count: expenses.length,
        sum: getExpensesTotal(expenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);