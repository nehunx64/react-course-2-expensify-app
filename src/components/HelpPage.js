import React from 'react';

const HelpPage = () => (
    <div>
        This is from my <strong>Help</strong> page
    </div>
);

export default HelpPage;