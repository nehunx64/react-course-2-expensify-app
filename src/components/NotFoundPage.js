import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <h1>404</h1> 
        <p>Resource not found!</p>
        <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPage;
