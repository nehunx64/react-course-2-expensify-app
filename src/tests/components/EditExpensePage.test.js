import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expensesfixtures from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    params = { id: expensesfixtures[1].id};
    wrapper = shallow(<EditExpensePage 
        expense={expensesfixtures[1]} 
        editExpense={editExpense} 
        removeExpense={removeExpense}
        history={history} />);
});

test('should render editExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle removeExpense', () => {
    wrapper.find('button').simulate('Click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({id: expensesfixtures[1].id});
});

test('should handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expensesfixtures[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expensesfixtures[1].id, expensesfixtures[1]);
});