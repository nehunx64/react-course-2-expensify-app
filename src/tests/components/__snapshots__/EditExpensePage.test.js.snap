// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 280,
        "createdAt": -345600000,
        "description": "plastic bags",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
